lib/sequent/core/aggregate_repository.rb

```diff
@@ -60,12 +60,43 @@ def load_aggregate(aggregate_id, clazz = nil)
           stream, events = @event_store.load_events(aggregate_id)
           raise AggregateNotFound.new(aggregate_id) unless stream
           aggregate_class = Class.const_get(stream.aggregate_type)
-          aggregates[aggregate_id] = aggregate_class.load_from_history(stream, events)
+          aggregate_class.load_from_history(stream, events)
         end
 
-        raise TypeError, "#{result.class} is not a #{clazz}" if result && clazz && !(result.class <= clazz)
+        raise TypeError, "#{result.class} is not a #{clazz}" if clazz && result && !(result.class <= clazz)
+        aggregates[aggregate_id] = result
+      end
+
+      ##
+      # Loads multiple aggregates at once.
+      # Returns the ones in the current Unit Of Work otherwise loads it from history.
+      #
+      # Note: This will load all the aggregates in memory, so querying 100s of aggregates
+      # with 100s of events could cause memory issues.
+      #
+      # Only returns the aggregates that exists.
+      # If +clazz+ is given and one of the aggregates is not of the correct type
+      # a +TypeError+ is raised.
+      #
+      # +aggregate_ids+ The ids of the aggregates to be loaded
+      # +clazz+ Optional argument that checks if all aggregates are of type +clazz+
+      def load_aggregates(aggregate_ids, clazz = nil)
```

Comment:
Simply delegate `load_aggregate` to this method by wrapping `aggregate_id` in an array?